core/src/main/java/com/linecorp/armeria/client/PreTlsHandshakeException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import com.linecorp.armeria.common.Flags;
+
+/**
+ * A {@link RuntimeException} raised before starting a TLS handshake.
+ */
+public final class PreTlsHandshakeException extends RuntimeException {
```

Comment:
Question) Would it make sense to extend from SslException since:
- Users can more handle expected exceptions more easily
```
catch(SslException e) {
...
}
```
- Although the exception occurs before a handshake, I think the exception is still pertinent to SSL.

Reply:
That's a good suggestion. 👍